sympy/geometry/parabola.py

```diff
@@ -268,6 +269,56 @@ def focus(self):
         """
         return self.args[0]
 
+    def intersection(self, o):
+        """The intersection of the parabola and another geometrical entity `o`.
+
+        Parameters
+        ==========
+
+        o : GeometryEntity, LinearEntity
+
+        Returns
+        =======
+
+        intersection : list of GeometryEntity objects
+
+        Examples
+        ========
+
+        >>> from sympy import Parabola, Point2D, Ellipse, Line, Segment2D
+        >>> p1 = Point2D(0,0)
+        >>> l1 = Line(Point2D(1, -2), Point2D(-1,-2))
+        >>> parabola1 = Parabola(p1, l1)
+        >>> parabola1.intersection(Ellipse(Point2D(0, 0), 2, 5))
+        [Point2D(-2, 0), Point2D(2, 0)]
+        >>> parabola1.intersection(Line(Point2D(-7, 3), Point2D(12, 3)))
+        [Point2D(-4, 3), Point2D(4, 3)]
+        >>> parabola1.intersection(Segment2D((-12, -65), (14, -68)))
+        []
+
+        """
+        x, y = symbols('x y', real=True)
+        parabola_eq = self.equation()
+        if isinstance(o, Parabola):
+            if o in self:
+                return [o]
+            else:
+                return [Point(i) for i in solve([parabola_eq, o.equation()], [x, y])]
+        elif isinstance(o, Point2D):
+            if simplify(parabola_eq.subs(([(x, o._args[0]), (y, o._args[1])]))) == 0:
+                return [o]
+            else:
+                return []
+        elif isinstance(o, (Segment2D, Ray2D)):
+            result = solve([parabola_eq, Line2D(o.points[0], o.points[1]).equation()], [x, y])
+            return [Point2D(i) for i in result if i in o]
```

Comment:
> PR #12350 has a version of nonlinsolve_real that could be used in your PR

Great! I thought about the same, but my concern was that the method `nonlinsolve_real` should also filter solution for positive dimensional problem. I am not sure, if your approach can do that? (I mean `number_hasimaginary(n)`) 
I see that you test only case, when solutions are numbers, but what happens, if we would like to solve system of equations, where as a result, we get for example complex plane? 

> Here and below, make the returned result canonical with list(ordered(foo)) where foo is your list.

I will do that. 
> The only suggestion that I would make for your PR is to use solveset_real or nonlinsolve instead of solve.

I think that using `solveset_real` is not convenient, beacuse it solves one equattion, but if you merged PR #12350 I will use in my PR `nonlinsolve_real` instead of `solve`.